src/options.c

```diff
@@ -54,6 +54,31 @@ int opt_meth_getoption(lua_State *L, p_opt opt, p_socket ps)
     return opt->func(L, ps);
 }
 
+/*------------------------------------------------------*/
+/* binds socket to network interface */
+int opt_set_bindtodevice(lua_State *L, p_socket ps)
+{
+#ifdef __APPLE__
+    return luaL_error(L, "SO_BINDTODEVICE is not supported on this operating system");
+#else
+    const char *dev = luaL_checkstring(L, 3);
+    return opt_set(L, ps, SOL_SOCKET, SO_BINDTODEVICE, (char*)dev, strlen(dev)+1);
+#endif
+}
+
+int opt_get_bindtodevice(lua_State *L, p_socket ps)
+{
+#ifdef __APPLE__
+    return luaL_error(L, "SO_BINDTODEVICE is not supported on this operating system");
+#else
+    char dev[IFNAMSIZ];
+    int len = sizeof(dev);
+    opt_get(L, ps, SOL_SOCKET, SO_BINDTODEVICE, &dev, &len);
+    lua_pushstring(L, dev);
```

Comment:
Not `checkstring()` directly as the function does not take parameters, but I in fact missed returning the error here in case of a failure. Fixed in 7084840.

Reply:
I'm sorry that was a finger-fart, `s/string/stack/`. What I meant to ask was `checkstack()`, in this case with just a value of 1 for the entry you're about to allocate:

```suggestion
    lua_checkstack(L, 1)
    lua_pushstring(L, dev);
```

Reply:
Ah, I see!
I can add it if preferred, but considering that the stack should always have 20 free slots whenever Lua invokes native code[[1]](http://www.lua.org/manual/5.1/manual.html#3.2) and the function pushes at max two items (the string `dev` or in case of errors, `nil` and the error), there shouldn't be room for problems.

Sorry for the late reply! Work came along.